src/globals.d

```diff
@@ -14,26 +14,50 @@ import ddmd.root.array;
 import ddmd.root.filename;
 import ddmd.root.outbuffer;
 
-template xversion(string s)
-{
-    enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
-}
-
 private string stripRight(string s)
 {
     while (s.length && (s[$ - 1] == ' ' || s[$ - 1] == '\n' || s[$ - 1] == '\r'))
         s = s[0 .. $ - 1];
     return s;
 }
 
-enum IN_GCC     = xversion!`IN_GCC`;
+immutable bool TARGET_LINUX, TARGET_OSX, TARGET_FREEBSD, TARGET_OPENBSD,
+               TARGET_SOLARIS, TARGET_WINDOS, TARGET_POSIX;
 
-enum TARGET_LINUX   = xversion!`linux`;
-enum TARGET_OSX     = xversion!`OSX`;
-enum TARGET_FREEBSD = xversion!`FreeBSD`;
-enum TARGET_OPENBSD = xversion!`OpenBSD`;
-enum TARGET_SOLARIS = xversion!`Solaris`;
-enum TARGET_WINDOS  = xversion!`Windows`;
+shared static this()
```

Comment:
The first step is to move all uses of `TARGET_XXX` into the `ddmd.target` module.  Second step is to make it private.  Third step is to find an alternative way to implement it.

But then again if all uses of `TARGET_XXX` are in the target module then whether or not step three is needed depends on the backend maintainers preference.

Regardless of what you think about cross compiling and how broken it is (why does ldc have all these problems that gdc doesn't? Cross compiling to both a different platform and architecture has worked for years), once it's in Target, there should be no issues. You *do* have your own local implementation of Target, right? Otherwise you have other problems that we can't help you with.